Userland/DevTools/UserspaceEmulator/SoftCPU.cpp

```diff
@@ -1855,9 +1855,12 @@ void SoftCPU::FLD_RM80(const X86::Instruction& insn)
     // long doubles can be up to 128 bits wide in memory for reasons (alignment) and only uses 80 bits of precision
     // GCC uses 12 bytes in 32 bit and 16 bytes in 64 bit mode
     // so in the 32 bit case we read a bit to much, but that shouldn't be an issue.
-    auto new_f80 = insn.modrm().read128(*this, insn);
     // FIXME: Respect shadow values
-    fpu_push(*(long double*)new_f80.value().bytes());
+    auto new_f80 = insn.modrm().read128(*this, insn).value();
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wstrict-aliasing"
```

Comment:
I cry :(
implement it in `AK/ByteReader.h` with the unions maybe?

Reply:
this is mainly needed for reasons described above, and maybe because the u128 class is not trivial enough...

Idk. why it worked with the previous implementation. though, because it was pretty much the same thing

Reply:
this is mainly needed for reasons described above, and maybe because the u128 class is not trivial enough...

Idk. why it worked with the previous implementation. though, because it was pretty much the same thing

Reply:
I think I wasn't clear enough on what I mean, you shouldn't be breaking strict aliasing here.

Reply:
Since the thing is a value, you can just use `bit_cast<long double>(that_value)`

Reply:
bit_cast asserts that the sizes are the same, which is not guaranteed in this case

Reply:
then just implement it in `ByteReader.h` (you'll see how it's done if you look at it) and use that.

Reply:
I just did it with a similar hack to the previous implementation...

PS:  
I think it wouldn't have been in `ByteReader.h`, but in `SoftMMU.h`, `Region.h`, `SimpleRegion.h` and `MMapRegion.h`, which is a bit annoying....